flexget/plugins/filter/unique.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+
+from flexget import plugin
+from flexget.event import event
+
+log = logging.getLogger('unique')
+
+
+class Unique(object):
+    """
+    Take action on entries with duplicate fields, except for the first item
+
+    Reject the second+ instance of every movie:
+
+      unique:
+        field:
+          - imdb_id
+          - movie_name
+        action: reject
+    """
+
+    schema = {
+        'oneOf': [
+            {'type': 'boolean'},
+            {
+                'type': 'object',
+                'properties': {
+                    'field': {
+                        'oneOf': [
+                            {'type': 'string'},
+                            {'type': 'array'},
+                        ]
+                    },
+                    'action': {'enum': ['accept', 'reject']},
+                },
+                'required': [],
+                'additionalProperties': False
+            },
+        ],
+    }
+
+    def prepare_config(self, config):
+        if isinstance(config, bool) or config is None:
+            config = {}
+        if 'action' not in config:
+            config['action'] = 'reject'
+        if 'field' not in config:
+            config['field']=[]
+        return config
+
+    def extract_fields(self, entry, field_names):
+        if field_names:
+            pass
+            return {field: entry.get(field) for field in field_names}
+        else:
+            return entry
+
+    def should_ignore(self, item, action):
+        return item.accepted and action == 'accept' or\
```

Comment:
Isn't this function only used in one place? Might make more sense to scrap the function then... It only makes it harder to read.